// @matchain/mid/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var index_exports = {};
__export(index_exports, {
  Components: () => components_exports,
  Hooks: () => hooks_exports,
  MatchProvider: () => MatchProvider,
  useMatch: () => useMatch
});
module.exports = __toCommonJS(index_exports);

// src/MatchContext.tsx
var import_react8 = require("react");

// src/assets/icon/ArrowLeftIcon.tsx
var import_jsx_runtime = require("react/jsx-runtime");
function ArrowLeftIcon({
  color = "#000",
  width = 24,
  height = 25,
  ...props
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
    "svg",
    {
      width,
      height,
      ...props,
      viewBox: "0 0 24 25",
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg",
      children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
        "path",
        {
          fillRule: "evenodd",
          clipRule: "evenodd",
          d: "M10.2322 5.26777L11.1161 6.15165L5.55207 11.7157H20.2157L20.2157 12.962L5.55207 12.962L11.1161 18.526L10.2322 19.4099L3.16117 12.3388L10.2322 5.26777Z",
          fill: color
        }
      )
    }
  );
}

// src/assets/icon/BackIcon.tsx
var import_jsx_runtime2 = require("react/jsx-runtime");

// src/assets/icon/CloseIcon.tsx
var import_jsx_runtime3 = require("react/jsx-runtime");

// src/assets/icon/XIcon.tsx
var import_jsx_runtime4 = require("react/jsx-runtime");
function XIcon({ size = 40 }) {
  return /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("svg", { width: size, height: size, viewBox: "0 0 40 40", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
    /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("rect", { width: "40", height: "40", rx: "8", fill: "black" }),
    /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
      "path",
      {
        d: "M16 10H9L17.2609 21.0145L9.44995 29.9999H12.1L18.4883 22.651L24 30H31L22.3917 18.5223L29.8001 10H27.1501L21.1643 16.8858L16 10ZM25 28L13 12H15L27 28H25Z",
        fill: "white"
      }
    )
  ] });
}

// src/assets/icon/GoogleIcon.tsx
var import_jsx_runtime5 = require("react/jsx-runtime");
function GoogleIcon({ size = 40 }) {
  return /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)("svg", { width: size, height: size, viewBox: "0 0 40 40", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
    /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("rect", { width: "40", height: "40", rx: "8", fill: "#F8F8F8" }),
    /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
      "path",
      {
        fillRule: "evenodd",
        clipRule: "evenodd",
        d: "M31.52 20.2729C31.52 19.422 31.4436 18.6038 31.3018 17.8184H20V22.4602H26.4582C26.18 23.9602 25.3345 25.2311 24.0636 26.082V29.0929H27.9418C30.2109 27.0038 31.52 23.9274 31.52 20.2729Z",
        fill: "#4285F4"
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
      "path",
      {
        fillRule: "evenodd",
        clipRule: "evenodd",
        d: "M20 32.0003C23.24 32.0003 25.9564 30.9257 27.9418 29.093L24.0636 26.0821C22.9891 26.8021 21.6145 27.2275 20 27.2275C16.8745 27.2275 14.2291 25.1166 13.2855 22.2803H9.27637V25.3894C11.2509 29.3112 15.3091 32.0003 20 32.0003Z",
        fill: "#34A853"
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
      "path",
      {
        fillRule: "evenodd",
        clipRule: "evenodd",
        d: "M13.2855 22.2804C13.0455 21.5604 12.9091 20.7913 12.9091 20.0004C12.9091 19.2095 13.0455 18.4404 13.2855 17.7204V14.6113H9.27636C8.46364 16.2313 8 18.0641 8 20.0004C8 21.9368 8.46364 23.7695 9.27636 25.3895L13.2855 22.2804Z",
        fill: "#FBBC05"
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
      "path",
      {
        fillRule: "evenodd",
        clipRule: "evenodd",
        d: "M20 12.7727C21.7618 12.7727 23.3436 13.3782 24.5873 14.5673L28.0291 11.1255C25.9509 9.18909 23.2345 8 20 8C15.3091 8 11.2509 10.6891 9.27637 14.6109L13.2855 17.72C14.2291 14.8836 16.8745 12.7727 20 12.7727Z",
        fill: "#EA4335"
      }
    )
  ] });
}

// src/assets/icon/WalletIcon.tsx
var import_jsx_runtime6 = require("react/jsx-runtime");
function WalletIcon({
  size = 40
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime6.jsxs)("svg", { width: size, height: size, viewBox: "0 0 40 40", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
    /* @__PURE__ */ (0, import_jsx_runtime6.jsx)("rect", { width: "40", height: "40", rx: "8", fill: "url(#paint0_linear_412_1774)" }),
    /* @__PURE__ */ (0, import_jsx_runtime6.jsx)(
      "path",
      {
        d: "M32.3335 19.1112V17.3333C32.3335 15.1242 30.5426 13.3333 28.3335 13.3333H25.6668M32.3335 19.1112H25.1112C23.2089 19.1112 21.6668 20.6533 21.6668 22.5556C21.6668 24.4579 23.2089 26 25.1112 26H32.3335M32.3335 19.1112C33.0699 19.1112 33.6668 19.7081 33.6668 20.4445V24.6667C33.6668 25.403 33.0699 26 32.3335 26M32.3335 26V28C32.3335 30.2091 30.5426 32 28.3335 32H11.0002C8.79102 32 7.00016 30.2091 7.00016 28V17.3333C7.00016 15.5917 8.11324 14.1101 9.66683 13.5609M25.6668 13.3333H11.0002C10.5326 13.3333 10.0839 13.4135 9.66683 13.5609M25.6668 13.3333V11.75C25.6668 9.91548 23.8575 8.62884 22.1247 9.2311L9.66683 13.5609",
        stroke: "white",
        strokeWidth: "1.33333",
        strokeLinecap: "round",
        strokeLinejoin: "round"
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime6.jsx)("circle", { cx: "26.1668", cy: "22.8335", r: "2", fill: "white" }),
    /* @__PURE__ */ (0, import_jsx_runtime6.jsx)("defs", { children: /* @__PURE__ */ (0, import_jsx_runtime6.jsxs)("linearGradient", { id: "paint0_linear_412_1774", x1: "20", y1: "0", x2: "20", y2: "40", gradientUnits: "userSpaceOnUse", children: [
      /* @__PURE__ */ (0, import_jsx_runtime6.jsx)("stop", { stopColor: "#FFB950" }),
      /* @__PURE__ */ (0, import_jsx_runtime6.jsx)("stop", { offset: "0.475", stopColor: "#FF7A60" }),
      /* @__PURE__ */ (0, import_jsx_runtime6.jsx)("stop", { offset: "1", stopColor: "#FF3C4A" })
    ] }) })
  ] });
}

// src/assets/icon/TelegramIcon.tsx
var import_jsx_runtime7 = require("react/jsx-runtime");
function TelegramIcon({
  size = 40
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime7.jsxs)("svg", { width: size, height: size, viewBox: "0 0 40 40", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
    /* @__PURE__ */ (0, import_jsx_runtime7.jsx)("rect", { width: "40", height: "40", rx: "8", fill: "#28A7E8" }),
    /* @__PURE__ */ (0, import_jsx_runtime7.jsx)(
      "path",
      {
        fillRule: "evenodd",
        clipRule: "evenodd",
        d: "M9.65904 18.6099C16.1373 15.7874 20.4571 13.9267 22.6186 13.0277C28.7899 10.4608 30.0723 10.0149 30.9081 10.0002C31.0919 9.99692 31.503 10.0425 31.7692 10.2585C31.994 10.4409 32.0559 10.6874 32.0855 10.8603C32.1151 11.0333 32.1519 11.4273 32.1226 11.7351C31.7882 15.249 30.3411 23.7762 29.605 27.7118C29.2935 29.3771 28.6801 29.9354 28.0863 29.9901C26.7958 30.1088 25.8159 29.1372 24.566 28.3179C22.6101 27.0358 21.5052 26.2377 19.6067 24.9867C17.4127 23.5408 18.835 22.7462 20.0853 21.4475C20.4126 21.1076 26.0985 15.9358 26.2085 15.4667C26.2223 15.408 26.2351 15.1893 26.1051 15.0738C25.9752 14.9583 25.7834 14.9978 25.645 15.0292C25.4489 15.0737 22.3244 17.1389 16.2715 21.2247C15.3847 21.8337 14.5814 22.1304 13.8616 22.1149C13.0682 22.0977 11.5419 21.6663 10.4073 21.2974C9.0156 20.8451 7.90953 20.6059 8.00585 19.8376C8.05601 19.4374 8.60708 19.0282 9.65904 18.6099Z",
        fill: "white"
      }
    )
  ] });
}

// src/assets/icon/EmailIcon.tsx
var import_jsx_runtime8 = require("react/jsx-runtime");
function EmailIcon({
  size = 40
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)("svg", { xmlns: "http://www.w3.org/2000/svg", width: "40", height: "40", viewBox: "0 0 40 40", fill: "none", children: [
    /* @__PURE__ */ (0, import_jsx_runtime8.jsx)("rect", { width: "40", height: "40", rx: "8", fill: "url(#paint0_linear_124_10994)" }),
    /* @__PURE__ */ (0, import_jsx_runtime8.jsx)(
      "path",
      {
        fillRule: "evenodd",
        clipRule: "evenodd",
        d: "M30.8867 12.5448C30.9585 12.4862 31 12.3997 31 12.3085C31 12.1381 30.8581 12 30.683 12H9.49205C9.2203 12 9 12.2144 9 12.4789C9 12.6212 9.06508 12.7562 9.17759 12.8472L19.5082 21.2008C19.7333 21.3828 20.06 21.3821 20.2842 21.199L30.8867 12.5448ZM9.99246 15.2938C9.59792 14.9748 9 15.2478 9 15.7471V27.9107C9 28.2362 9.27109 28.5 9.6055 28.5H30.3945C30.7289 28.5 31 28.2362 31 27.9107V15.5103C31 15.0098 30.3994 14.7372 30.0054 15.0588L20.353 22.9375C20.0642 23.1732 19.7375 23.1739 19.5124 22.9919L9.99246 15.2938Z",
        fill: "white"
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime8.jsx)("defs", { children: /* @__PURE__ */ (0, import_jsx_runtime8.jsxs)("linearGradient", { id: "paint0_linear_124_10994", x1: "20", y1: "0", x2: "20", y2: "40", gradientUnits: "userSpaceOnUse", children: [
      /* @__PURE__ */ (0, import_jsx_runtime8.jsx)("stop", { stopColor: "#FFB950" }),
      /* @__PURE__ */ (0, import_jsx_runtime8.jsx)("stop", { offset: "0.475", stopColor: "#FF7A60" }),
      /* @__PURE__ */ (0, import_jsx_runtime8.jsx)("stop", { offset: "1", stopColor: "#FF3C4A" })
    ] }) })
  ] });
}

// src/assets/icon/ArrowRightIcon.tsx
var import_jsx_runtime9 = require("react/jsx-runtime");
function ArrowRightIcon({
  color = "#000",
  size = 20,
  className,
  style
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime9.jsx)("svg", { className, style, width: size, height: size, viewBox: "0 0 20 20", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ (0, import_jsx_runtime9.jsx)(
    "path",
    {
      fillRule: "evenodd",
      clipRule: "evenodd",
      d: "M11.4733 3.97314L10.7367 4.70971L15.3734 9.34641L3.15374 9.34641L3.15374 10.385L15.3734 10.385L10.7367 15.0217L11.4733 15.7583L17.3658 9.8657L11.4733 3.97314Z",
      fill: color
    }
  ) });
}

// src/assets/icon/CloseRoundIcon.tsx
var import_jsx_runtime10 = require("react/jsx-runtime");
function CloseRoundIcon({ size = 30, ...props }) {
  return /* @__PURE__ */ (0, import_jsx_runtime10.jsxs)("svg", { width: size, height: size, ...props, viewBox: "0 0 30 30", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)("rect", { width: "30", height: "30", rx: "15", fill: "#F9F9F9" }),
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)("path", { d: "M10.7574 10.7573L19.2426 19.2426", stroke: "#6E6E6E", strokeWidth: "1.5", strokeLinecap: "round" }),
    /* @__PURE__ */ (0, import_jsx_runtime10.jsx)("path", { d: "M19.2426 10.7573L10.7574 19.2426", stroke: "#6E6E6E", strokeWidth: "1.5", strokeLinecap: "round" })
  ] });
}

// src/assets/icon/LoadingIcon.tsx
var import_jsx_runtime11 = require("react/jsx-runtime");
var LoadingIcon = ({ size = 22, className = "", color = "white" }) => /* @__PURE__ */ (0, import_jsx_runtime11.jsxs)(
  "svg",
  {
    width: size,
    height: size,
    viewBox: "0 0 22 22",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    className,
    children: [
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)("path", { d: "M11 1V3.5", stroke: color, strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)("path", { opacity: "0.3", d: "M3.92834 3.92871L5.69977 5.70017", stroke: color, strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)("path", { opacity: "0.4", d: "M1 11H3.5", stroke: color, strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)("path", { opacity: "0.5", d: "M3.92834 18.0722L5.69977 16.3008", stroke: color, strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)("path", { opacity: "0.6", d: "M11 21V18.5", stroke: color, strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)("path", { opacity: "0.7", d: "M18.0715 18.0722L16.3 16.3008", stroke: color, strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)("path", { opacity: "0.8", d: "M21 11H18.5", stroke: color, strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }),
      /* @__PURE__ */ (0, import_jsx_runtime11.jsx)("path", { opacity: "0.9", d: "M18.0715 3.92871L16.3 5.70017", stroke: color, strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" })
    ]
  }
);
var LoadingIcon_default = LoadingIcon;

// src/assets/icon/EmailLineIcon.tsx
var import_jsx_runtime12 = require("react/jsx-runtime");
function EmailLineIcon({
  size = 24,
  color = "black",
  ...props
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime12.jsx)("svg", { width: size, height: size, ...props, viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ (0, import_jsx_runtime12.jsx)(
    "path",
    {
      d: "M3 8.00049L10.8906 13.2609C11.5624 13.7088 12.4376 13.7088 13.1094 13.2609L21 8.00049M5 19.0005H19C20.1046 19.0005 21 18.1051 21 17.0005V7.00049C21 5.89592 20.1046 5.00049 19 5.00049H5C3.89543 5.00049 3 5.89592 3 7.00049V17.0005C3 18.1051 3.89543 19.0005 5 19.0005Z",
      stroke: color,
      strokeWidth: "1.5",
      strokeLinecap: "round",
      strokeLinejoin: "round"
    }
  ) });
}

// src/assets/icon/UnLoginIcon.tsx
var import_jsx_runtime13 = require("react/jsx-runtime");
var UnLoginIcon = ({ size = 32, color = "white", ...props }) => /* @__PURE__ */ (0, import_jsx_runtime13.jsx)("svg", { xmlns: "http://www.w3.org/2000/svg", width: size, height: size, viewBox: "0 0 32 32", fill: "none", ...props, children: /* @__PURE__ */ (0, import_jsx_runtime13.jsx)(
  "path",
  {
    d: "M24 14L26 12M24 14L22 12M24 14L22 16M24 14L26 16M18 12C18 14.2091 16.2091 16 14 16C11.7909 16 10 14.2091 10 12C10 9.79086 11.7909 8 14 8C16.2091 8 18 9.79086 18 12ZM10 19H18C20.2091 19 22 20.7909 22 23C22 23.5523 21.5523 24 21 24H7C6.44772 24 6 23.5523 6 23C6 20.7909 7.79086 19 10 19Z",
    stroke: color,
    strokeWidth: "1.5",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }
) });
var UnLoginIcon_default = UnLoginIcon;

// src/assets/icon/LoginIcon.tsx
var import_jsx_runtime14 = require("react/jsx-runtime");
var LoginIcon = ({ size = 32, color = "white", ...props }) => /* @__PURE__ */ (0, import_jsx_runtime14.jsxs)("svg", { width: size, height: size, viewBox: "0 0 32 32", xmlns: "http://www.w3.org/2000/svg", ...props, children: [
  /* @__PURE__ */ (0, import_jsx_runtime14.jsx)("rect", { width: "32", height: "32", rx: "16" }),
  /* @__PURE__ */ (0, import_jsx_runtime14.jsx)(
    "path",
    {
      d: "M22.5 14L24 15.5L27 12.5M19 11C19 13.2091 17.2091 15 15 15C12.7909 15 11 13.2091 11 11C11 8.79086 12.7909 7 15 7C17.2091 7 19 8.79086 19 11ZM11 18H19C21.2091 18 23 19.7909 23 22C23 22.5523 22.5523 23 22 23H8C7.44772 23 7 22.5523 7 22C7 19.7909 8.79086 18 11 18Z",
      stroke: "white",
      strokeWidth: "1.5",
      strokeLinecap: "round",
      strokeLinejoin: "round"
    }
  )
] });
var LoginIcon_default = LoginIcon;

// src/assets/icon/CheckRoundIcon.tsx
var import_jsx_runtime15 = require("react/jsx-runtime");
function CheckRoundIcon({
  size,
  color = "#2CBF68",
  ...props
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime15.jsxs)("svg", { width: size, height: size, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props, children: [
    /* @__PURE__ */ (0, import_jsx_runtime15.jsxs)("g", { clipPath: "url(#clip0_418_7699)", children: [
      /* @__PURE__ */ (0, import_jsx_runtime15.jsx)(
        "path",
        {
          d: "M14.6668 8.00016C14.6668 11.6821 11.6821 14.6668 8.00016 14.6668C4.31826 14.6668 1.3335 11.6821 1.3335 8.00016C1.3335 4.31826 4.31826 1.3335 8.00016 1.3335C11.6821 1.3335 14.6668 4.31826 14.6668 8.00016Z",
          fill: color,
          stroke: color,
          strokeLinecap: "round"
        }
      ),
      /* @__PURE__ */ (0, import_jsx_runtime15.jsx)(
        "path",
        {
          d: "M10.6867 5.97994C10.882 6.1752 10.882 6.49179 10.6867 6.68705L7.35339 10.0204C7.15813 10.2156 6.84155 10.2156 6.64628 10.0204L5.31295 8.68705C5.11769 8.49179 5.11769 8.1752 5.31295 7.97994C5.50821 7.78468 5.8248 7.78468 6.02006 7.97994L6.99984 8.95972L8.48973 7.46983L9.97962 5.97994C10.1749 5.78468 10.4915 5.78468 10.6867 5.97994Z",
          fill: "white"
        }
      )
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime15.jsx)("defs", { children: /* @__PURE__ */ (0, import_jsx_runtime15.jsx)("clipPath", { id: "clip0_418_7699", children: /* @__PURE__ */ (0, import_jsx_runtime15.jsx)("rect", { width: "16", height: "16", fill: "white" }) }) })
  ] });
}

// src/assets/icon/PasswordRoundIcon.tsx
var import_jsx_runtime16 = require("react/jsx-runtime");
function PasswordRoundIcon() {
  return /* @__PURE__ */ (0, import_jsx_runtime16.jsxs)("svg", { width: "40", height: "40", viewBox: "0 0 40 40", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: [
    /* @__PURE__ */ (0, import_jsx_runtime16.jsx)("rect", { x: "0.5", y: "0.5", width: "39", height: "39", rx: "7.5", fill: "white" }),
    /* @__PURE__ */ (0, import_jsx_runtime16.jsx)("rect", { x: "0.5", y: "0.5", width: "39", height: "39", rx: "7.5", stroke: "black" }),
    /* @__PURE__ */ (0, import_jsx_runtime16.jsx)(
      "path",
      {
        fillRule: "evenodd",
        clipRule: "evenodd",
        d: "M19.7673 16.1137C19.962 13.9479 21.7831 12.25 24 12.25C26.3472 12.25 28.25 14.1528 28.25 16.5C28.25 18.8472 26.3472 20.75 24 20.75C23.488 20.75 22.9987 20.6598 22.5461 20.495C21.9175 20.2662 21.1115 20.3279 20.5486 20.8908L19.2626 22.1768C19.2157 22.2237 19.1521 22.25 19.0858 22.25H18C17.3096 22.25 16.75 22.8096 16.75 23.5V24.25H15.5C14.8096 24.25 14.25 24.8096 14.25 25.5V26.5C14.25 26.6381 14.1381 26.75 14 26.75H12C11.8619 26.75 11.75 26.6381 11.75 26.5V23.9538C11.75 23.8817 11.7811 23.8131 11.8354 23.7656L19.1524 17.3632C19.5404 17.0238 19.7275 16.556 19.7673 16.1137ZM24 10.75C20.9997 10.75 18.5369 13.0473 18.2733 15.9794C18.2614 16.1115 18.2107 16.1941 18.1646 16.2344L10.8476 22.6368C10.4678 22.9691 10.25 23.4491 10.25 23.9538V26.5C10.25 27.4665 11.0335 28.25 12 28.25H14C14.9665 28.25 15.75 27.4665 15.75 26.5V25.75H17C17.6904 25.75 18.25 25.1904 18.25 24.5V23.75H19.0858C19.5499 23.75 19.995 23.5656 20.3232 23.2374L21.6092 21.9514C21.6738 21.8868 21.8277 21.8298 22.033 21.9045C22.6476 22.1283 23.3104 22.25 24 22.25C27.1756 22.25 29.75 19.6756 29.75 16.5C29.75 13.3244 27.1756 10.75 24 10.75ZM25 16.5C25.5523 16.5 26 16.0523 26 15.5C26 14.9477 25.5523 14.5 25 14.5C24.4477 14.5 24 14.9477 24 15.5C24 16.0523 24.4477 16.5 25 16.5Z",
        fill: "black"
      }
    )
  ] });
}

// src/assets/icon/CloseEyeIcon.tsx
var import_jsx_runtime17 = require("react/jsx-runtime");
function CloseEyeIcon() {
  return /* @__PURE__ */ (0, import_jsx_runtime17.jsx)("svg", { width: "20", height: "20", viewBox: "0 0 20 20", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ (0, import_jsx_runtime17.jsx)(
    "path",
    {
      d: "M9.15132 12.5568V11.5464C8.21031 11.4949 7.28023 11.3193 6.38524 11.024L6.04233 11.9579C6.00626 12.0669 5.94867 12.1676 5.87299 12.2539C5.79731 12.3403 5.70508 12.4106 5.60174 12.4607C5.4984 12.5108 5.38607 12.5396 5.27139 12.5455C5.15671 12.5514 5.04201 12.5342 4.93409 12.4949C4.82617 12.4557 4.72722 12.3952 4.6431 12.317C4.55898 12.2389 4.49139 12.1446 4.44435 12.0399C4.3973 11.9351 4.37175 11.822 4.36921 11.7072C4.36667 11.5924 4.38719 11.4783 4.42955 11.3715L4.80788 10.3268C4.04603 9.88461 3.32634 9.37345 2.65789 8.79975C2.54358 8.70488 2.42928 8.61687 2.32184 8.52772C2.1442 8.38554 2.03031 8.17862 2.00523 7.95247C1.98015 7.72633 2.04593 7.49948 2.18811 7.32184C2.33028 7.1442 2.5372 7.03031 2.76335 7.00523C2.98949 6.98015 3.21634 7.04593 3.39398 7.18811C3.50828 7.27955 3.62258 7.37099 3.73689 7.47043C5.46065 9.00826 7.68996 9.85815 10 9.85815C12.31 9.85815 14.5393 9.00826 16.2631 7.47043C16.3774 7.37099 16.4917 7.27955 16.606 7.18811C16.694 7.11771 16.7949 7.06533 16.9031 7.03395C17.0114 7.00257 17.1247 6.99281 17.2367 7.00523C17.3486 7.01765 17.4571 7.052 17.5558 7.10632C17.6545 7.16065 17.7415 7.23388 17.8119 7.32184C17.8823 7.4098 17.9347 7.51077 17.9661 7.61897C17.9974 7.72717 18.0072 7.8405 17.9948 7.95247C17.9824 8.06445 17.948 8.17288 17.8937 8.27158C17.8394 8.37028 17.7661 8.45732 17.6782 8.52772L17.3421 8.79975C16.6737 9.37354 15.954 9.88471 15.1921 10.3268L15.5705 11.3715C15.6128 11.4783 15.6333 11.5924 15.6308 11.7072C15.6283 11.822 15.6027 11.9351 15.5557 12.0399C15.5086 12.1446 15.441 12.2389 15.3569 12.317C15.2728 12.3952 15.1738 12.4557 15.0659 12.4949C14.958 12.5342 14.8433 12.5514 14.7286 12.5455C14.6139 12.5396 14.5016 12.5108 14.3983 12.4607C14.2949 12.4106 14.2027 12.3403 14.127 12.2539C14.0513 12.1676 13.9937 12.0669 13.9577 11.9579L13.6148 11.024C12.7198 11.3193 11.7897 11.4949 10.8487 11.5464V12.5568C10.8487 12.7842 10.7584 13.0022 10.5976 13.163C10.4368 13.3238 10.2188 13.4141 9.99143 13.4141C9.76407 13.4141 9.54602 13.3238 9.38526 13.163C9.22449 13.0022 9.13417 12.7842 9.13417 12.5568H9.15132Z",
      fill: "#6E6E6E"
    }
  ) });
}

// src/assets/icon/OpenEyeIcon.tsx
var import_jsx_runtime18 = require("react/jsx-runtime");
function OpenEyeIcon() {
  return /* @__PURE__ */ (0, import_jsx_runtime18.jsx)("svg", { width: "20", height: "20", viewBox: "0 0 20 20", fill: "none", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ (0, import_jsx_runtime18.jsx)(
    "path",
    {
      d: "M10 6.4543C11.3373 6.44986 12.6486 6.82287 13.7834 7.53047C14.9181 8.23806 15.8301 9.25151 16.4145 10.4543C15.2145 12.9052 12.7564 14.4543 10 14.4543C7.24364 14.4543 4.78545 12.9052 3.58545 10.4543C4.1699 9.25151 5.08191 8.23806 6.21664 7.53047C7.35136 6.82287 8.66274 6.44986 10 6.4543ZM10 4.99976C6.36364 4.99976 3.25818 7.26157 2 10.4543C3.25818 13.647 6.36364 15.9088 10 15.9088C13.6364 15.9088 16.7418 13.647 18 10.4543C16.7418 7.26157 13.6364 4.99976 10 4.99976ZM10 8.63612C10.4822 8.63617 10.9446 8.82774 11.2855 9.16871C11.6265 9.50968 11.818 9.97212 11.818 10.4543C11.818 10.9365 11.6265 11.3989 11.2855 11.7399C10.9446 12.0809 10.4822 12.2724 10 12.2725C9.51782 12.2724 9.0554 12.0809 8.71447 11.7399C8.37353 11.3989 8.18199 10.9365 8.18199 10.4543C8.18199 9.97212 8.37353 9.50968 8.71447 9.16871C9.0554 8.82774 9.51782 8.63617 10 8.63612ZM10 7.18157C8.19636 7.18157 6.72727 8.65066 6.72727 10.4543C6.72727 12.2579 8.19636 13.727 10 13.727C11.8036 13.727 13.2727 12.2579 13.2727 10.4543C13.2727 8.65066 11.8036 7.18157 10 7.18157Z",
      fill: "#6E6E6E"
    }
  ) });
}

// src/assets/icon/DeleteRoundIcon.tsx
var import_jsx_runtime19 = require("react/jsx-runtime");
function DeleteRoundIcon({ height = 21, width = 20, color = "var(--matchid-error-color)", ...props }) {
  return /* @__PURE__ */ (0, import_jsx_runtime19.jsxs)("svg", { width, height, viewBox: "0 0 20 21", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props, children: [
    /* @__PURE__ */ (0, import_jsx_runtime19.jsx)("rect", { y: "0.5", width: "20", height: "20", rx: "10", fill: color }),
    /* @__PURE__ */ (0, import_jsx_runtime19.jsx)("path", { d: "M7.17139 7.67188L12.8282 13.3287", stroke: "white", strokeLinecap: "round" }),
    /* @__PURE__ */ (0, import_jsx_runtime19.jsx)("path", { d: "M12.8286 7.67188L7.17176 13.3287", stroke: "white", strokeLinecap: "round" })
  ] });
}

// src/components/Modal/index.tsx
var import_jsx_runtime20 = require("react/jsx-runtime");
function Modal({
  children,
  isOpen,
  width = 480
}) {
  return isOpen ? /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(
    "div",
    {
      className: "matchid-overlay",
      children: /* @__PURE__ */ (0, import_jsx_runtime20.jsx)("div", { className: "matchid-modal", style: {
        width
      }, children })
    }
  ) : /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(import_jsx_runtime20.Fragment, {});
}
function ModalWithHeader({
  children,
  onBack,
  onClose,
  title,
  showClose = true,
  ...props
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime20.jsxs)(Modal, { ...props, children: [
    /* @__PURE__ */ (0, import_jsx_runtime20.jsxs)("div", { className: `matchid-modal-header`, children: [
      /* @__PURE__ */ (0, import_jsx_runtime20.jsxs)("div", { className: "matchid-modal-header-content", children: [
        onBack && /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(ArrowLeftIcon, { className: "matchid-modal-header-back", onClick: onBack }),
        /* @__PURE__ */ (0, import_jsx_runtime20.jsx)("span", { className: "matchid-modal-header-title", children: title })
      ] }),
      onClose && /* @__PURE__ */ (0, import_jsx_runtime20.jsx)(CloseRoundIcon, { className: "matchid-modal-header-close", onClick: onClose })
    ] }),
    children
  ] });
}

// src/components/PasswordModal/index.tsx
var import_react5 = require("react");

// src/components/Input/index.tsx
var import_react = require("react");
var import_jsx_runtime21 = require("react/jsx-runtime");
function Input({
  onChange,
  type,
  after,
  ...props
}) {
  const [inputType, setInputType] = (0, import_react.useState)(type);
  return /* @__PURE__ */ (0, import_jsx_runtime21.jsxs)("div", { className: `matchid-input-box ${props.value.length > 0 ? "matchid-input-has-content" : ""}`, children: [
    /* @__PURE__ */ (0, import_jsx_runtime21.jsx)("input", { type: inputType, onChange, ...props, className: "matchid-input-field" }),
    props.value.length > 0 && /* @__PURE__ */ (0, import_jsx_runtime21.jsx)(DeleteRoundIcon, { onClick: (e) => {
      if (onChange) {
        onChange({ target: { value: "" } });
      }
    }, className: "matchid-input-delete-icon", color: "var(--matchid-input-delete-icon-color)" }),
    type === "password" && /* @__PURE__ */ (0, import_jsx_runtime21.jsx)("div", { className: "matchid-input-eye-icon", onClick: () => {
      setInputType(inputType === "password" ? "text" : "password");
    }, children: inputType === "password" ? /* @__PURE__ */ (0, import_jsx_runtime21.jsx)(CloseEyeIcon, {}) : /* @__PURE__ */ (0, import_jsx_runtime21.jsx)(OpenEyeIcon, {}) }),
    after
  ] });
}

// src/components/Field/index.tsx
var import_jsx_runtime22 = require("react/jsx-runtime");
function Field({
  label,
  children,
  error
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime22.jsxs)("div", { className: "matchid-field-box", children: [
    /* @__PURE__ */ (0, import_jsx_runtime22.jsx)("div", { className: "matchid-field-label", children: label }),
    children,
    error && /* @__PURE__ */ (0, import_jsx_runtime22.jsx)("div", { className: "matchid-field-error", children: error })
  ] });
}

// src/components/Button/index.tsx
var import_jsx_runtime23 = require("react/jsx-runtime");
function Button({
  size = "df",
  disabled = false,
  loading = false,
  children,
  onClick,
  highlight = false,
  block = false,
  type = "button",
  rounded = true,
  className = "",
  style = {}
}) {
  const onAction = () => {
    if (!disabled && !loading) {
      onClick && onClick();
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime23.jsx)(
    "button",
    {
      type,
      className: `${className} matchid-btn ${"matchid-btn-" + size} ${highlight ? "matchid-btn-highlight" : ""} ${loading ? "matchid-btn-loading" : ""}  ${block ? "matchid-btn-block" : ""}  ${rounded ? "matchid-btn-rounded" : ""}`,
      disabled: disabled || loading,
      style: {
        ...style
      },
      onClick: onAction,
      children: loading ? /* @__PURE__ */ (0, import_jsx_runtime23.jsx)(LoadingIcon_default, { className: "matchid-btn-loading-icon", color: "var(--matchid-btn-loading-color)" }) : children
    }
  );
}

// src/hooks/index.tsx
var hooks_exports = {};
__export(hooks_exports, {
  useMatchEvents: () => useMatchEvents,
  useUserInfo: () => useUserInfo,
  useWallet: () => useWallet
});

// src/store/useLocalStore.ts
var import_zustand = require("zustand");
var import_middleware = require("zustand/middleware");
var persistedState = (0, import_middleware.persist)(
  (set) => ({
    appid: "",
    token: "",
    did: "",
    env: "main",
    mid: "",
    overview: null,
    theme: "light",
    setOverview: (overview) => set({
      overview,
      address: overview.address,
      did: overview.did,
      mid: overview.mid
    }),
    setEnv: (env) => set({ env }),
    setDid: (did) => set({ did }),
    setToken: (token) => set({ token }),
    setAppid: (appid) => set({ appid }),
    setMid: (mid) => set({ mid }),
    logout: () => set({ token: "", did: "", mid: "", address: "", overview: null }),
    setTheme: (theme) => set({ theme }),
    address: "",
    setAddress: (address) => set({ address })
  }),
  { name: "match-local" }
);
var useLocalStore = (0, import_zustand.create)((0, import_middleware.devtools)(persistedState));
var localStore = useLocalStore;
var useLocalStore_default = useLocalStore;

// src/hooks/useUserInfo.tsx
var import_react2 = require("react");

// src/api/request.ts
var import_axios = __toESM(require("axios"));

// src/config/env/main.ts
var main_default = {
  endpoints: {
    back: "https://api.matchid.ai/mid/api/v1/",
    login: "https://auth.matchid.ai/"
  }
};

// src/config/env/dev.ts
var dev_default = {
  endpoints: {
    back: "https://mid-test-api.matchain.io/mid/api/v1/",
    login: "https://calm-hare-roughly.ngrok-free.app/"
  }
};

// src/config/env/test.ts
var test_default = {
  endpoints: {
    back: "https://mid-test-api.matchain.io/mid/api/v1/",
    login: "https://mid-sso-test.vercel.app/"
  }
};

// src/config/env/index.ts
var env_default = {
  main: main_default,
  dev: dev_default,
  test: test_default
};

// src/store/index.ts
var getEnv = () => {
  try {
    const store = localStore.getState();
    const env = store?.env;
    if (env) {
      return env;
    } else {
      return "main";
    }
  } catch (e) {
    return "main";
  }
};
var getEnvConfig = () => {
  return env_default[getEnv()];
};
var getEndpoints = () => {
  return getEnvConfig().endpoints;
};
var getAppid = () => {
  try {
    const store = localStore.getState();
    const appid = store?.appid;
    if (appid) {
      return appid;
    } else {
      return "";
    }
  } catch (e) {
    return "";
  }
};
var getToken = () => {
  try {
    const store = localStore.getState();
    const token = store?.token;
    if (token) {
      return token;
    } else {
      return "";
    }
  } catch (e) {
    return "";
  }
};

// src/api/request.ts
var SUCCESS_CODE = 0;
var isSuccess = (res) => {
  return res.code === SUCCESS_CODE;
};
var instance = import_axios.default.create({
  timeout: 6e4,
  validateStatus(status) {
    return status >= 200 && status <= 500;
  }
});
var request = async (config) => {
  try {
    const endpoints = getEndpoints();
    const token = getToken();
    instance.defaults.baseURL = `${endpoints.back}`;
    instance.defaults.headers.common["Appid"] = getAppid();
    if (token) {
      instance.defaults.headers.common["Authorization"] = token;
    }
    const { data } = await instance.request(config);
    console.log("api", data);
    if (data.code == 401001) {
      localStore.getState().logout();
      throw new Error("Your session has expired, please log in again");
    }
    return data;
  } catch (err) {
    console.error("qwe-err", err);
    const message = "Request Error";
    console.error(message);
    return {
      code: -1,
      data: null,
      message,
      success: false,
      fail: true,
      result: null
    };
  }
};
var request_default = request;

// src/api/index.ts
var getEmailCodeApi = (email) => {
  return request_default({
    url: `/email/code`,
    method: "POST",
    data: { email }
  });
};
var verifyEmailCodeApi = ({ email, verification_key, verification_code }) => {
  return request_default({
    url: `/user/email/login`,
    method: "POST",
    data: { email, verification_key, verification_code }
  });
};
var getOverviewInfoApi = () => {
  return request_default({
    url: `/user/overview`,
    method: "GET"
  });
};
var toLogoutApi = () => {
  return request_default({
    url: `/user/logout`,
    method: "POST"
  });
};
var setUserNameApi = (data) => {
  return request_default({
    url: `/user/name`,
    method: "POSt",
    data
  });
};

// src/hooks/eventManager.ts
var EventManager = class {
  constructor() {
    this.listeners = {};
  }
  on(event, callback) {
    if (!this.listeners[event]) {
      this.listeners[event] = /* @__PURE__ */ new Set();
    }
    this.listeners[event].add(callback);
  }
  off(event, callback) {
    if (this.listeners[event]) {
      this.listeners[event].delete(callback);
      if (this.listeners[event].size === 0) {
        delete this.listeners[event];
      }
    }
  }
  emit(event, ...args) {
    if (this.listeners[event]) {
      this.listeners[event].forEach((callback) => callback(...args));
    }
  }
};
var eventManager = new EventManager();
var eventManager_default = eventManager;

// src/utils/index.ts
var getVersion = () => {
  return "0.0.1";
};
function encodeBase64(input) {
  return btoa(unescape(encodeURIComponent(input)));
}
var isValidEmail = (email) => {
  return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
};
var isValidUsername = (username) => {
  return /^[a-zA-Z0-9_-]+$/.test(username);
};
function truncateAddress(address) {
  if (address.length <= 6) {
    return address;
  }
  const start = address.slice(0, 4);
  const end = address.slice(-2);
  return `${start}...${end}`;
}

// src/hooks/useUserInfo.tsx
function useUserInfo() {
  const {
    appid,
    token,
    mid,
    did,
    env,
    logout: logoutStore,
    setOverview,
    overview,
    address
  } = useLocalStore_default();
  const { events, login } = useMatch();
  const endpoints = (0, import_react2.useMemo)(() => {
    return getEndpoints();
  }, [env]);
  const isLogin = (0, import_react2.useMemo)(() => !!token, [token]);
  const logout = async () => {
    try {
      await toLogoutApi();
    } catch (err) {
      console.error("logout", err);
    }
    logoutStore();
    events && events.onLogout && events.onLogout();
    eventManager_default.emit("onLogout");
  };
  const loginByMethod = (method) => {
    const link = `${endpoints.back}auth/${method}?appid=${appid}&provider=${method}&redirect=${encodeURIComponent(endpoints.login + "login/" + method)}&authorization=${(/* @__PURE__ */ new Date()).getTime()}`;
    console.log("link", link);
    const authWindow = window.open(
      link,
      // Replace with the actual authorization URL
      "authWindow",
      "width=800,height=600"
    );
  };
  const loginByTwitter = () => {
    loginByMethod("twitter");
  };
  const loginByGoogle = () => {
    loginByMethod("google");
  };
  const getLoginEmailCode = async (email) => {
    const res = await getEmailCodeApi(email);
    if (res && res.data && isSuccess(res)) {
      window.sessionStorage.setItem("loginkey", res.data.key);
      return res.data.key;
    }
    return "";
  };
  const loginByEmail = async ({
    email,
    code
  }) => {
    try {
      const obj = {
        email,
        verification_key: window.sessionStorage.getItem("loginkey"),
        verification_code: code
      };
      const res = await verifyEmailCodeApi(obj);
      if (res && res.data && isSuccess(res)) {
        await login({
          mid: res.data.mid,
          token: `${res.data.token_type} ${res.data.access_token}`
        });
        return true;
      } else {
        throw new Error(res.message);
      }
    } catch (error) {
      console.error("loginByEmail", error);
      throw error;
    }
    return false;
  };
  const loginByWallet = () => {
    const authWindow = window.open(
      `${endpoints.login}login/wallet?appid=${appid}`,
      // Replace with the actual authorization URL
      "_blank"
      // 'width=800,height=600'
    );
  };
  const loginByTelegram = () => {
    const authWindow = window.open(
      `${endpoints.login}login/telegram?appid=${appid}`,
      // Replace with the actual authorization URL
      "authWindow",
      "width=800,height=600"
    );
  };
  const refreshOverview = async () => {
    const res = await getOverviewInfoApi();
    if (res.data) {
      setOverview(res.data);
    }
  };
  const bindWallet = async () => {
    if (!token) {
      throw new Error("You must login first");
    }
    const authWindow = window.open(
      `${endpoints.login}bind/wallet?appid=${appid}&st=` + encodeURIComponent(encodeBase64(token)),
      // Replace with the actual authorization URL
      "_blank"
    );
  };
  const bindTelegram = async () => {
    if (!token) {
      throw new Error("You must login first");
    }
    const authWindow = window.open(
      `${endpoints.login}bind/telegram?appid=${appid}&st=` + encodeURIComponent(encodeBase64(token)),
      // Replace with the actual authorization URL
      "authWindow",
      "width=800,height=600"
    );
  };
  return {
    loginByTelegram,
    loginByTwitter,
    loginByGoogle,
    loginByWallet,
    loginByEmail,
    token,
    mid,
    did,
    address,
    isLogin,
    logout,
    getLoginEmailCode,
    refreshOverview,
    overview,
    bindWallet,
    bindTelegram,
    username: overview?.username || ""
  };
}

// src/hooks/useMatchEvents.ts
var import_react3 = require("react");
function useMatchEvents(handlers) {
  (0, import_react3.useEffect)(() => {
    Object.entries(handlers).forEach(([event, handler]) => {
      if (handler) {
        eventManager_default.on(event, handler);
      }
    });
    return () => {
      Object.entries(handlers).forEach(([event, handler]) => {
        if (handler) {
          eventManager_default.off(event, handler);
        }
      });
    };
  }, [handlers]);
}

// src/hooks/useWallet.tsx
var import_react4 = require("react");

// src/store/useStore.ts
var import_zustand2 = require("zustand");
var useStore = (0, import_zustand2.create)((set) => ({
  walletIframeInited: false,
  setWalletIframeInited: (inited) => set({ walletIframeInited: inited }),
  recoveryModal: {
    open: false,
    success: async () => {
    },
    close: async () => {
    }
  },
  setRecoveryModal: (params) => set({ recoveryModal: params }),
  closeRecoveryModal: () => set({ recoveryModal: { open: false } })
}));
var useStore_default = useStore;

// src/hooks/useWallet.tsx
var AppClientId = "react-sdk-" + getVersion();
function useWallet() {
  const { appid, token, overview } = useLocalStore_default();
  const { setRecoveryModal, closeRecoveryModal } = useStore_default();
  const initWallet = ({
    did,
    address
  }) => {
    return window.waitMatchUntilWalletMessage(
      "initCore",
      {
        AppId: appid,
        AppClientId,
        UserId: did,
        AccessToken: token,
        Address: address
      }
    );
  };
  const isRecovered = async () => {
    const res = await window.waitMatchUntilWalletMessage("isRecovered");
    return res.isRecovered;
  };
  const generateWallet = async ({
    did,
    userPasscode
  }) => {
    return await window.waitMatchUntilWalletMessage("generateWallet", {
      recoveryType: "user_passcode_recovery_key",
      userPasscode
    });
  };
  const signMessage = async (message, type) => {
    return new Promise(async (resolve, reject) => {
      const res = await isRecovered();
      console.log("isRecovered", res);
      const getSign = async () => {
        try {
          const recoverRes = await window.waitMatchUntilWalletMessage("signMessage", {
            message,
            chainType: type
          });
          closeRecoveryModal();
          resolve(recoverRes);
        } catch (error) {
          console.error("qwe-sign-error", error);
          reject(error);
        }
      };
      if (!res) {
        setRecoveryModal({
          open: true,
          success: async () => {
            console.log("recover Success");
            await getSign();
          },
          close: async () => {
            closeRecoveryModal();
            reject(new Error("recover modal close"));
          }
        });
        await initWallet({
          address: overview?.address || "",
          did: overview?.did?.split(":")[2] || ""
        });
      } else {
        getSign();
      }
    });
  };
  const signTransaction = async (transaction, type) => {
    return new Promise(async (resolve, reject) => {
      const res = await isRecovered();
      console.log("isRecovered", res);
      const getSign = async () => {
        try {
          const recoverRes = await window.waitMatchUntilWalletMessage("signTransaction", {
            transaction,
            chainType: type
          });
          closeRecoveryModal();
          resolve(recoverRes);
        } catch (error) {
          console.error("qwe-sign-error", error);
          reject(error);
        }
      };
      if (!res) {
        setRecoveryModal({
          open: true,
          success: async () => {
            console.log("recover Success");
            await getSign();
          },
          close: async () => {
            closeRecoveryModal();
            reject(new Error("recover modal close"));
          }
        });
        await initWallet({
          address: overview?.address || "",
          did: overview?.did?.split(":")[2] || ""
        });
      } else {
        getSign();
      }
    });
  };
  const recoveryWallet = async (chainType, recoveryType, userPasscode) => {
    return await window.waitMatchUntilWalletMessage("recoveryWallet", {
      chainType,
      recoveryType,
      userPasscode
    });
  };
  return {
    recoveryWallet,
    initWallet,
    generateWallet,
    signMessage,
    signTransaction,
    isRecovered
  };
}
var useWalletInit = ({
  refreshOverview
}) => {
  const { env } = useLocalStore_default();
  const getWalletIframe = () => {
    return document.getElementById("match-wallet");
  };
  const [walletInited, setWalletInited] = (0, import_react4.useState)(false);
  const { appid, token, overview } = useLocalStore_default();
  const { initWallet, generateWallet } = useWallet();
  (0, import_react4.useEffect)(() => {
    if (env) {
      if (!window.matchWalletMessageIdMap) {
        window.matchWalletMessageIdMap = {};
      }
      const config = env_default[env];
      const endpoints = config.endpoints;
      const existingIframe = getWalletIframe();
      if (!existingIframe) {
        const iframe = document.createElement("iframe");
        iframe.id = "match-wallet";
        iframe.src = endpoints.login + "wallet";
        iframe.style.display = "none";
        iframe.style.width = "0";
        iframe.style.height = "0";
        iframe.onload = () => {
          setWalletInited(true);
        };
        document.body.insertBefore(iframe, document.body.firstChild);
        window.sendMatchWalletMessage = (method, data) => {
          const messageId = Date.now().toString() + Math.random().toString().slice(6);
          const message = {
            method,
            data,
            messageId,
            source: "matchid"
          };
          if (document.getElementById("match-wallet")) {
            getWalletIframe()?.contentWindow?.postMessage(message, "*");
            return messageId;
          }
          console.error("not found iframe");
          return false;
        };
        window.waitMatchUntilWalletMessage = async (method, data, timeout = 5e3) => {
          return new Promise((resolve, reject) => {
            const messageId = window.sendMatchWalletMessage(method, data);
            if (!messageId) {
              console.error("Can't find wallet message");
              reject(new Error("Can't find wallet message"));
              return;
            }
            window.matchWalletMessageIdMap[messageId] = { resolve, reject };
            window.matchWalletMessageIdMap[messageId].timeout = setTimeout(() => {
              console.error("Get wallet result timeout", {
                messageId,
                method,
                data
              });
              delete window.matchWalletMessageIdMap[messageId];
              reject(new Error("Get wallet result timeout"));
            }, timeout);
          });
        };
      } else {
        if (existingIframe.src !== endpoints.login + "wallet") {
          existingIframe.src = endpoints.login + "wallet";
        }
        setWalletInited(true);
      }
    }
  }, [env]);
  (0, import_react4.useEffect)(() => {
    const messageHandle = async (e) => {
      const res = e.data;
      if (res.source != "match-wallet") {
        return;
      }
      const messageId = res.messageId;
      if (messageId && window.matchWalletMessageIdMap[messageId]) {
        const { resolve, reject, timeout } = window.matchWalletMessageIdMap[messageId];
        delete window.matchWalletMessageIdMap[messageId];
        clearTimeout(timeout);
        if (res.status == "success") {
          resolve(res.data);
        } else {
          console.error("qwe-waller-error", res);
          reject(new Error(res.data.message));
        }
      }
    };
    window.addEventListener("message", messageHandle);
    return () => {
      window.removeEventListener("message", messageHandle);
    };
  }, []);
  (0, import_react4.useEffect)(() => {
    if (token && overview && overview.did && walletInited) {
      const did = overview.did.split(":")[2];
      const newUserInit = async () => {
        await window.waitMatchUntilWalletMessage(
          "initCore",
          {
            AppId: appid,
            AppClientId,
            UserId: did,
            AccessToken: token,
            Address: ""
          }
        );
      };
      if (!overview.address) {
        newUserInit();
      } else {
        initWallet({
          address: overview.address,
          did
        });
      }
    }
  }, [overview, token, walletInited]);
  return {
    walletInited
  };
};

// src/components/PasswordModal/index.tsx
var import_jsx_runtime24 = require("react/jsx-runtime");
function PasswordModal({
  title,
  isOpen,
  onSuccess,
  ...props
}) {
  const { refreshOverview } = useUserInfo();
  const { isLogin, did } = useUserInfo();
  const [password, setPassword] = (0, import_react5.useState)("");
  const [rePassword, setRePassword] = (0, import_react5.useState)("");
  const [error, setError] = (0, import_react5.useState)("");
  const { generateWallet } = useWallet();
  const passwordError = (0, import_react5.useMemo)(() => {
    if (password.length < 6) return "Password must be at least 6 characters";
    return "";
  }, [password]);
  const rePasswordError = (0, import_react5.useMemo)(() => {
    if (rePassword != password) {
      return "The password you entered twice do not match";
    }
    return "";
  }, [rePassword, password]);
  (0, import_react5.useEffect)(() => {
    if (isOpen) {
      setPassword("");
      setRePassword("");
    }
  }, [isOpen]);
  const [isSubmitting, setIsSubmitting] = (0, import_react5.useState)(false);
  const onContinue = async () => {
    if (isSubmitting) return;
    try {
      setIsSubmitting(true);
      const address = await generateWallet({
        did: did.split(":")[2],
        userPasscode: password
      });
      await refreshOverview();
      onSuccess && onSuccess();
    } catch (error2) {
      setError(error2.message);
    } finally {
      setIsSubmitting(false);
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime24.jsx)(ModalWithHeader, { isOpen: isOpen && isLogin, ...props, title: title || "Set Password", children: /* @__PURE__ */ (0, import_jsx_runtime24.jsxs)("div", { className: "matchid-password-box", children: [
    /* @__PURE__ */ (0, import_jsx_runtime24.jsxs)("div", { className: "matchid-password-header", children: [
      /* @__PURE__ */ (0, import_jsx_runtime24.jsx)("div", { className: "matchid-password-header-icon", children: /* @__PURE__ */ (0, import_jsx_runtime24.jsx)(PasswordRoundIcon, {}) }),
      /* @__PURE__ */ (0, import_jsx_runtime24.jsx)("div", { className: "matchid-password-header-content", children: "Please set the wallet password that will be used to recover the wallet" })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime24.jsxs)("div", { className: "matchid-password-content", children: [
      /* @__PURE__ */ (0, import_jsx_runtime24.jsx)(Field, { label: "Password", error: password.length > 0 && passwordError, children: /* @__PURE__ */ (0, import_jsx_runtime24.jsx)(
        Input,
        {
          placeholder: "Enter the Password",
          maxLength: 32,
          type: "password",
          onChange: (e) => setPassword(e.target.value),
          value: password
        }
      ) }),
      /* @__PURE__ */ (0, import_jsx_runtime24.jsx)(Field, { label: "Re Password", error: rePassword.length > 0 ? rePasswordError || error : error, children: /* @__PURE__ */ (0, import_jsx_runtime24.jsx)(
        Input,
        {
          placeholder: "Re Enter the Password",
          maxLength: 32,
          onChange: (e) => setRePassword(e.target.value),
          value: rePassword,
          type: "password"
        }
      ) })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime24.jsx)(
      Button,
      {
        disabled: password.length == 0 || !!passwordError || !!rePasswordError,
        highlight: true,
        block: true,
        size: "lg",
        onClick: onContinue,
        loading: isSubmitting,
        children: "Continue"
      }
    )
  ] }) });
}

// src/components/RecoveryModal/index.tsx
var import_react6 = require("react");
var import_jsx_runtime25 = require("react/jsx-runtime");
function RecoveryModal({
  title,
  isOpen,
  onSuccess,
  ...props
}) {
  const { refreshOverview } = useUserInfo();
  const { isLogin } = useUserInfo();
  const [password, setPassword] = (0, import_react6.useState)("");
  const { recoveryWallet } = useWallet();
  const [error, setError] = (0, import_react6.useState)("");
  const passwordError = (0, import_react6.useMemo)(() => {
    if (password.length < 6) return "Password must be at least 6 characters";
    return "";
  }, [password]);
  (0, import_react6.useEffect)(() => {
    if (isOpen) {
      setPassword("");
    }
  }, [isOpen]);
  const [isSubmitting, setIsSubmitting] = (0, import_react6.useState)(false);
  const onContinue = async () => {
    if (isSubmitting) return;
    try {
      setIsSubmitting(true);
      await recoveryWallet("ethereum", "user_passcode_recovery_key", password);
      await refreshOverview();
      onSuccess && onSuccess();
    } catch (error2) {
      console.error("qwe-recover-error", error2);
      setError(error2.message);
    } finally {
      setIsSubmitting(false);
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime25.jsx)(ModalWithHeader, { isOpen: isOpen && isLogin, ...props, title: title || "Recover Wallet", children: /* @__PURE__ */ (0, import_jsx_runtime25.jsxs)("div", { className: "matchid-password-box", children: [
    /* @__PURE__ */ (0, import_jsx_runtime25.jsxs)("div", { className: "matchid-password-header", children: [
      /* @__PURE__ */ (0, import_jsx_runtime25.jsx)("div", { className: "matchid-password-header-icon", children: /* @__PURE__ */ (0, import_jsx_runtime25.jsx)(PasswordRoundIcon, {}) }),
      /* @__PURE__ */ (0, import_jsx_runtime25.jsx)("div", { className: "matchid-password-header-content", children: "Please enter your password to recover your wallet" })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime25.jsx)("div", { className: "matchid-password-content", children: /* @__PURE__ */ (0, import_jsx_runtime25.jsx)(Field, { label: "Password", error: password.length > 0 ? passwordError || error : error, children: /* @__PURE__ */ (0, import_jsx_runtime25.jsx)(
      Input,
      {
        placeholder: "Enter the Password",
        maxLength: 32,
        type: "password",
        onChange: (e) => setPassword(e.target.value),
        value: password
      }
    ) }) }),
    /* @__PURE__ */ (0, import_jsx_runtime25.jsx)(
      Button,
      {
        disabled: password.length == 0 || !!passwordError,
        highlight: true,
        block: true,
        size: "lg",
        onClick: onContinue,
        loading: isSubmitting,
        children: "Continue"
      }
    )
  ] }) });
}

// src/context/BusinessProvider.tsx
var import_jsx_runtime26 = require("react/jsx-runtime");
function BusinessProvider({ children }) {
  const { overview, token } = useUserInfo();
  const { recoveryModal } = useStore_default();
  return /* @__PURE__ */ (0, import_jsx_runtime26.jsxs)(import_jsx_runtime26.Fragment, { children: [
    /* @__PURE__ */ (0, import_jsx_runtime26.jsx)(PasswordModal, { isOpen: !!token && !!overview && !overview.address && !!overview.did, showClose: false }),
    /* @__PURE__ */ (0, import_jsx_runtime26.jsx)(
      RecoveryModal,
      {
        isOpen: !!token && !!overview && !!overview.address && recoveryModal.open,
        onClose: recoveryModal.close,
        onSuccess: recoveryModal.success
      }
    ),
    children
  ] });
}

// src/context/index.tsx
var import_jsx_runtime27 = (
  // <MpcWalletProvider>
  require("react/jsx-runtime")
);
var Providers = ({ children }) => {
  return /* @__PURE__ */ (0, import_jsx_runtime27.jsx)(BusinessProvider, { children });
};
var context_default = Providers;

// src/hooks/useInit.tsx
var import_react7 = require("react");
function useInit({
  theme,
  appid,
  env,
  events
}) {
  const {
    setAppid,
    setEnv,
    setToken,
    setMid,
    token,
    setOverview,
    setTheme
  } = useLocalStore_default();
  const overviewLoadingRef = (0, import_react7.useRef)(false);
  const searchParams = new URLSearchParams(window.location.search);
  const matchToken = searchParams.get("matchToken");
  const config = env_default[env];
  const endpoints = config.endpoints;
  (0, import_react7.useEffect)(() => {
    setTheme(theme);
  }, [theme]);
  (0, import_react7.useEffect)(() => {
    setAppid(appid);
  }, [appid]);
  (0, import_react7.useEffect)(() => {
    setEnv(env);
  }, [env]);
  (0, import_react7.useEffect)(() => {
    if (matchToken) {
      const tokenData = JSON.parse(atob(matchToken));
      if (tokenData && tokenData.mid && tokenData.token) {
        login({
          mid: tokenData.mid,
          token: "Bearer " + tokenData.token
        });
      }
    }
  }, [matchToken]);
  (0, import_react7.useEffect)(() => {
    const onLoginMessage = (event) => {
      const res = event.data;
      if (res.event === "login" && res.data && (res.data.token || res.data.token_type && res.data.access_token)) {
        login({
          mid: res.data.mid,
          token: res.data.token || `${res.data.token_type} ${res.data.access_token}`
        });
      } else if (res.event == "bind") {
        eventManager_default.emit("onBind", res.data);
        if (events && events.onBind) {
          events.onBind(res.data);
        }
      }
    };
    window.addEventListener("message", onLoginMessage);
    return () => {
      window.removeEventListener("message", onLoginMessage);
    };
  }, []);
  const loadOverview = async () => {
    console.log("loadOverview");
    if (overviewLoadingRef.current) {
      return;
    }
    overviewLoadingRef.current = true;
    try {
      const res = await getOverviewInfoApi();
      setOverview(res.data);
      return res.data;
    } catch (err) {
      console.error("overview error", err);
    } finally {
      overviewLoadingRef.current = false;
    }
  };
  (0, import_react7.useEffect)(() => {
    if (token) {
      loadOverview();
    }
  }, [token]);
  const login = async ({ mid, token: token2 }) => {
    setMid(mid);
    setToken(token2);
    const overview = await loadOverview();
    eventManager_default.emit("onLogin", {
      mid,
      token: token2,
      did: overview?.did
    });
    if (events && events?.onLogin) {
      events.onLogin({
        mid,
        token: token2,
        did: overview?.did
      });
    }
  };
  return {
    loadOverview,
    login,
    config,
    endpoints
  };
}

// src/MatchContext.tsx
var import_jsx_runtime28 = require("react/jsx-runtime");
var MatchContext = (0, import_react8.createContext)(void 0);
var MatchProvider = ({ children, appid, env = "main", events, theme = "light" }) => {
  const { loadOverview, login, config, endpoints } = useInit({
    theme,
    appid,
    env,
    events
  });
  useWalletInit({
    refreshOverview: loadOverview
  });
  return /* @__PURE__ */ (0, import_jsx_runtime28.jsx)(
    MatchContext.Provider,
    {
      value: {
        appid,
        env,
        endpoints,
        events,
        login,
        theme
      },
      children: /* @__PURE__ */ (0, import_jsx_runtime28.jsx)(context_default, { children })
    }
  );
};
var useMatch = () => {
  const context = (0, import_react8.useContext)(MatchContext);
  if (context === void 0) {
    throw new Error("useMatch must be used within a MatchProvider");
  }
  return context;
};

// src/components/index.tsx
var components_exports = {};
__export(components_exports, {
  Button: () => Button,
  EmailModal: () => EmailModal,
  Field: () => Field,
  Input: () => Input,
  LoginBox: () => LoginBox,
  LoginButton: () => LoginButton,
  LoginModal: () => LoginModal,
  LoginPanel: () => LoginPanel,
  Modal: () => Modal,
  ModalWithHeader: () => ModalWithHeader,
  PasswordModal: () => PasswordModal,
  UsernameModal: () => UsernameModal
});

// src/components/EmailModal/index.tsx
var import_react11 = require("react");

// src/components/EmailModal/StepEmail.tsx
var import_react9 = require("react");
var import_jsx_runtime29 = require("react/jsx-runtime");
function StepEmail(props) {
  const [emailVal, setEmailVal] = (0, import_react9.useState)("");
  (0, import_react9.useEffect)(() => {
    if (props.email) {
      setEmailVal(props.email);
    }
  }, []);
  const canContinue = (0, import_react9.useMemo)(() => {
    return isValidEmail(emailVal);
  }, [emailVal]);
  const onContinue = async () => {
    props.onContinue(emailVal);
  };
  return /* @__PURE__ */ (0, import_jsx_runtime29.jsxs)("div", { className: "matchid-email-email-box", children: [
    /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(Field, { label: "Email Address", children: /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(
      Input,
      {
        placeholder: "Enter Your Email Address",
        onChange: (e) => setEmailVal(e.target.value),
        value: emailVal
      }
    ) }),
    /* @__PURE__ */ (0, import_jsx_runtime29.jsx)(Button, { disabled: !canContinue, style: {
      marginTop: "64px"
    }, onClick: onContinue, size: "lg", block: true, highlight: true, children: "Continue" })
  ] });
}

// src/components/EmailModal/StepVerify.tsx
var import_react10 = require("react");

// src/config/index.tsx
var EMAIL_INTERVAL = 60;
var EMAIL_CODE_LENGTH = 6;

// src/components/EmailModal/StepVerify.tsx
var import_jsx_runtime30 = require("react/jsx-runtime");
function StepVerify(props) {
  const { getLoginEmailCode, loginByEmail } = useUserInfo();
  const [error, setError] = (0, import_react10.useState)("");
  const [code, setCode] = (0, import_react10.useState)("");
  const [sending, setSending] = (0, import_react10.useState)(false);
  const [submitting, setSubmitting] = (0, import_react10.useState)(false);
  const sendTimeRef = (0, import_react10.useRef)(0);
  const [sendBtnText, setSendBtnText] = (0, import_react10.useState)("Send");
  const intervalTime = EMAIL_INTERVAL;
  const codeLength = EMAIL_CODE_LENGTH;
  const intervalRef = (0, import_react10.useRef)(null);
  const onSend = async () => {
    if (sendTimeRef.current > 0) {
      return;
    }
    try {
      setError("");
      setSending(true);
      sendTimeRef.current = intervalTime;
      await getLoginEmailCode(props.email);
      setSendBtnText(`${sendTimeRef.current}s`);
      intervalRef.current = setInterval(() => {
        sendTimeRef.current--;
        setSendBtnText(`${sendTimeRef.current}s`);
        if (sendTimeRef.current <= 0) {
          setSendBtnText("Resend");
          clearInterval(intervalRef.current);
          setSending(false);
        }
      }, 1e3);
    } catch (err) {
      console.error("Send email error", err);
      setError("Failed to send code:" + err.message);
      setSending(false);
    }
  };
  (0, import_react10.useEffect)(() => {
    onSend();
    return () => {
      if (intervalRef.current) {
        clearInterval(intervalRef.current);
      }
    };
  }, []);
  const canContinue = (0, import_react10.useMemo)(() => {
    return code.length === codeLength;
  }, [code]);
  const onContinue = async () => {
    if (submitting) {
      return;
    }
    try {
      setError("");
      setSubmitting(true);
      const res = await loginByEmail({
        email: props.email,
        code
      });
      if (res) {
        props.onSuccess && props.onSuccess();
      }
      setSubmitting(false);
    } catch (err) {
      console.error("Verify email code error", err);
      setError(err.message);
      setSubmitting(false);
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime30.jsxs)("div", { className: "matchid-email-verify-box", children: [
    /* @__PURE__ */ (0, import_jsx_runtime30.jsxs)("div", { className: "matchid-email-verify-header", children: [
      /* @__PURE__ */ (0, import_jsx_runtime30.jsx)("div", { className: "matchid-email-verify-header-icon", children: /* @__PURE__ */ (0, import_jsx_runtime30.jsx)(EmailLineIcon, {}) }),
      /* @__PURE__ */ (0, import_jsx_runtime30.jsxs)("div", { className: "matchid-email-verify-header-content", children: [
        /* @__PURE__ */ (0, import_jsx_runtime30.jsx)("div", { className: "matchid-email-verify-header-value", children: props.email }),
        /* @__PURE__ */ (0, import_jsx_runtime30.jsx)("div", { className: "matchid-email-verify-header-tips", children: "We have sent a verification code to your email" })
      ] })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime30.jsx)(Field, { label: "Verification Code", error, children: /* @__PURE__ */ (0, import_jsx_runtime30.jsx)(
      Input,
      {
        placeholder: "Enter the code",
        maxLength: codeLength,
        onChange: (e) => setCode(e.target.value),
        value: code,
        after: /* @__PURE__ */ (0, import_jsx_runtime30.jsx)(
          Button,
          {
            highlight: true,
            disabled: sending,
            style: {
              height: "100%",
              borderTopLeftRadius: 0,
              borderBottomLeftRadius: 0,
              borderTopRightRadius: "inherit",
              borderBottomRightRadius: "inherit",
              width: "80px"
            },
            onClick: onSend,
            children: sendBtnText
          }
        )
      }
    ) }),
    /* @__PURE__ */ (0, import_jsx_runtime30.jsx)(Button, { disabled: !canContinue, highlight: true, block: true, size: "lg", onClick: onContinue, children: "Continue" })
  ] });
}

// src/components/EmailModal/index.tsx
var import_jsx_runtime31 = require("react/jsx-runtime");
function EmailModal({
  isOpen = false,
  width = 480,
  onClose,
  onBack,
  onLogin
}) {
  const [step, setStep] = (0, import_react11.useState)("input");
  const [emailVal, setEmailVal] = (0, import_react11.useState)("");
  (0, import_react11.useEffect)(() => {
    if (!isOpen) {
      setStep("input");
      setEmailVal("");
    }
  }, [isOpen]);
  return /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(
    ModalWithHeader,
    {
      isOpen,
      width,
      onClose,
      title: "Email",
      onBack: step == "verify" ? () => setStep("input") : onBack,
      children: step === "input" ? /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(StepEmail, { email: emailVal, onContinue: (email) => {
        setEmailVal(email);
        setStep("verify");
      } }) : /* @__PURE__ */ (0, import_jsx_runtime31.jsx)(StepVerify, { email: emailVal, onSuccess: onLogin })
    }
  );
}

// src/components/LoginBox/index.tsx
var import_react12 = require("react");
var import_jsx_runtime32 = require("react/jsx-runtime");
function LoginBox({
  methods,
  inModal = false
}) {
  const [emailOpen, setEmailOpen] = (0, import_react12.useState)(false);
  const { loginByTelegram, loginByTwitter, loginByGoogle, loginByWallet } = useUserInfo();
  const defaultMethods = [
    "wallet",
    "google",
    "email",
    "telegram",
    "X"
  ];
  const methodList = (0, import_react12.useMemo)(() => {
    return methods || defaultMethods;
  }, [methods]);
  const LoginItem = ({
    icon,
    name,
    onClick
  }) => {
    return /* @__PURE__ */ (0, import_jsx_runtime32.jsxs)("div", { className: "matchid-login-method", onClick, children: [
      /* @__PURE__ */ (0, import_jsx_runtime32.jsxs)("div", { className: "matchid-login-method-content", children: [
        /* @__PURE__ */ (0, import_jsx_runtime32.jsx)("div", { className: "matchid-login-method-icon", children: icon }),
        /* @__PURE__ */ (0, import_jsx_runtime32.jsx)("span", { className: "matchid-login-method-name", children: name })
      ] }),
      /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(ArrowRightIcon, { className: "matchid-login-method-arrow", size: 20, color: "var(--matchid-arrow-color)" })
    ] });
  };
  const methodMap = {
    wallet: {
      icon: /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(WalletIcon, {}),
      name: "Wallet",
      onClick: loginByWallet
    },
    email: {
      icon: /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(EmailIcon, {}),
      name: "Email",
      onClick: () => {
        setEmailOpen(true);
      }
    },
    google: {
      icon: /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(GoogleIcon, {}),
      name: "Google",
      onClick: loginByGoogle
    },
    X: {
      icon: /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(XIcon, {}),
      name: "X",
      onClick: loginByTwitter
    },
    telegram: {
      icon: /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(TelegramIcon, {}),
      name: "Telegram",
      onClick: loginByTelegram
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime32.jsxs)(import_jsx_runtime32.Fragment, { children: [
    (!inModal || !emailOpen) && /* @__PURE__ */ (0, import_jsx_runtime32.jsx)("div", { className: "matchid-login-box", children: methodList.map((m) => {
      return /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(
        LoginItem,
        {
          icon: methodMap[m].icon,
          name: methodMap[m].name,
          onClick: methodMap[m].onClick
        },
        m
      );
    }) }),
    /* @__PURE__ */ (0, import_jsx_runtime32.jsx)(
      EmailModal,
      {
        isOpen: emailOpen,
        onClose: () => {
          setEmailOpen(false);
        },
        onBack: inModal ? () => {
          setEmailOpen(false);
        } : void 0
      }
    )
  ] });
}

// src/components/LoginButton/index.tsx
var import_react13 = require("react");

// src/components/LoginPanel/index.tsx
var import_jsx_runtime33 = require("react/jsx-runtime");
function LoginPanel({
  header,
  onClose,
  ...props
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime33.jsxs)("div", { className: "matchid-login-panel", children: [
    header ? header : /* @__PURE__ */ (0, import_jsx_runtime33.jsxs)("div", { className: "matchid-login-panel-header", children: [
      /* @__PURE__ */ (0, import_jsx_runtime33.jsxs)("div", { className: "matchid-login-panel-header-content", children: [
        /* @__PURE__ */ (0, import_jsx_runtime33.jsx)("div", { className: "matchid-login-panel-header-title", children: "Log in / Sign up" }),
        /* @__PURE__ */ (0, import_jsx_runtime33.jsx)("div", { className: "matchid-login-panel-header-subtilte", children: "You can use the following methods" })
      ] }),
      onClose && /* @__PURE__ */ (0, import_jsx_runtime33.jsx)("div", { className: "matchid-login-panel-header-close", onClick: onClose, children: /* @__PURE__ */ (0, import_jsx_runtime33.jsx)(CloseRoundIcon, {}) })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime33.jsx)("div", { className: "matchid-login-panel-divide" }),
    /* @__PURE__ */ (0, import_jsx_runtime33.jsx)(LoginBox, { ...props })
  ] });
}

// src/components/LoginModal/index.tsx
var import_jsx_runtime34 = require("react/jsx-runtime");
function LoginModal({
  isOpen = false,
  width = 480,
  ...props
}) {
  const { isLogin } = useUserInfo();
  return /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(
    Modal,
    {
      isOpen: isOpen && !isLogin,
      width,
      children: /* @__PURE__ */ (0, import_jsx_runtime34.jsx)(LoginPanel, { ...props, inModal: true })
    }
  );
}

// src/components/LoginButton/index.tsx
var import_jsx_runtime35 = require("react/jsx-runtime");
function LoginButton({
  loginRender,
  methods,
  onLoginClick,
  ...props
}) {
  const { isLogin, username } = useUserInfo();
  const [loginOpen, setLoginOpen] = (0, import_react13.useState)(false);
  if (!isLogin) {
    return /* @__PURE__ */ (0, import_jsx_runtime35.jsxs)(import_jsx_runtime35.Fragment, { children: [
      /* @__PURE__ */ (0, import_jsx_runtime35.jsx)(LoginModal, { methods, isOpen: loginOpen, onClose: () => setLoginOpen(false) }),
      /* @__PURE__ */ (0, import_jsx_runtime35.jsxs)(Button, { className: "matchid-unlogin-btn", ...props, highlight: true, onClick: () => setLoginOpen(true), children: [
        /* @__PURE__ */ (0, import_jsx_runtime35.jsx)(UnLoginIcon_default, {}),
        /* @__PURE__ */ (0, import_jsx_runtime35.jsx)("span", { children: "Login" })
      ] })
    ] });
  }
  return loginRender ? loginRender() : /* @__PURE__ */ (0, import_jsx_runtime35.jsxs)(Button, { onClick: onLoginClick, className: "matchid-login-btn", ...props, children: [
    /* @__PURE__ */ (0, import_jsx_runtime35.jsx)(LoginIcon_default, {}),
    /* @__PURE__ */ (0, import_jsx_runtime35.jsx)("span", { children: username ? truncateAddress(username) : "MatchID User" })
  ] });
}

// src/components/UsernameModal/index.tsx
var import_react14 = require("react");

// src/assets/icon/InfoRoundIcon.tsx
var import_jsx_runtime36 = require("react/jsx-runtime");
function InfoRoundIcon({
  size,
  color = "#6E6E6E",
  ...props
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime36.jsxs)("svg", { width: size, height: size, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props, children: [
    /* @__PURE__ */ (0, import_jsx_runtime36.jsx)("g", { clipPath: "url(#clip0_418_7746)", children: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)(
      "path",
      {
        fillRule: "evenodd",
        clipRule: "evenodd",
        d: "M7.99984 0.666504C3.94975 0.666504 0.666504 3.94975 0.666504 7.99984C0.666504 12.0499 3.94975 15.3332 7.99984 15.3332C12.0499 15.3332 15.3332 12.0499 15.3332 7.99984C15.3332 3.94975 12.0499 0.666504 7.99984 0.666504ZM1.99984 7.99984C1.99984 4.68613 4.68613 1.99984 7.99984 1.99984C11.3135 1.99984 13.9998 4.68613 13.9998 7.99984C13.9998 11.3135 11.3135 13.9998 7.99984 13.9998C4.68613 13.9998 1.99984 11.3135 1.99984 7.99984ZM8.6665 7.33317C8.6665 6.96498 8.36803 6.6665 7.99984 6.6665C7.63165 6.6665 7.33317 6.96498 7.33317 7.33317V11.3332C7.33317 11.7014 7.63165 11.9998 7.99984 11.9998C8.36803 11.9998 8.6665 11.7014 8.6665 11.3332V7.33317ZM7.99984 5.33317C8.36803 5.33317 8.6665 5.03469 8.6665 4.6665C8.6665 4.29831 8.36803 3.99984 7.99984 3.99984C7.63165 3.99984 7.33317 4.29831 7.33317 4.6665C7.33317 5.03469 7.63165 5.33317 7.99984 5.33317Z",
        fill: color
      }
    ) }),
    /* @__PURE__ */ (0, import_jsx_runtime36.jsx)("defs", { children: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)("clipPath", { id: "clip0_418_7746", children: /* @__PURE__ */ (0, import_jsx_runtime36.jsx)("rect", { width: "16", height: "16", fill: "white" }) }) })
  ] });
}

// src/components/UsernameModal/index.tsx
var import_jsx_runtime37 = require("react/jsx-runtime");
var ValidItem = ({
  success = false,
  text
}) => {
  return /* @__PURE__ */ (0, import_jsx_runtime37.jsxs)("div", { className: `matchid-valid-status-item matchid-valid-status-${success ? "success" : "error"}`, children: [
    success ? /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(CheckRoundIcon, { size: 16 }) : /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(InfoRoundIcon, { size: 16 }),
    /* @__PURE__ */ (0, import_jsx_runtime37.jsx)("span", { children: text })
  ] });
};
function UsernameModal({
  title,
  isOpen,
  onSuccess,
  ...props
}) {
  const { username, refreshOverview } = useUserInfo();
  const { isLogin } = useUserInfo();
  const [val, setVal] = (0, import_react14.useState)(username);
  const [error, setError] = (0, import_react14.useState)("");
  (0, import_react14.useEffect)(() => {
    if (isOpen) {
      setVal(username);
      setError("");
    }
  }, [isOpen]);
  const isValid = (0, import_react14.useMemo)(() => {
    return isValidUsername(val);
  }, [val]);
  const isLength = (0, import_react14.useMemo)(() => {
    return val.length >= 2 && val.length <= 32;
  }, [val]);
  const isSafe = isValid && isLength;
  const [isSubmitting, setIsSubmitting] = (0, import_react14.useState)(false);
  const onSubmit = async () => {
    if (isSubmitting) return;
    try {
      setIsSubmitting(true);
      const res = await setUserNameApi({
        username: val
      });
      if (isSuccess(res)) {
        await refreshOverview();
        onSuccess && onSuccess();
      } else {
        setError(res.message);
      }
    } catch (error2) {
      setError(error2.message);
    } finally {
      setIsSubmitting(false);
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(ModalWithHeader, { isOpen: isOpen && isLogin, ...props, title: title || "Set User Name", children: /* @__PURE__ */ (0, import_jsx_runtime37.jsxs)("div", { className: "matchid-username-box", children: [
    /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(Field, { label: "User Name", error, children: /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(
      Input,
      {
        placeholder: "Enter Your User Name",
        onChange: (e) => {
          setVal(e.target.value);
          setError("");
        },
        value: val
      }
    ) }),
    /* @__PURE__ */ (0, import_jsx_runtime37.jsxs)("div", { className: "matchid-valid", children: [
      /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(
        ValidItem,
        {
          success: isValid,
          text: "Name can be composed of numbers and letters as well as characters"
        }
      ),
      /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(ValidItem, { success: isLength, text: "No less than 2 characters" })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(Button, { disabled: !isSafe, loading: isSubmitting, style: {
      marginTop: "64px"
    }, onClick: onSubmit, size: "lg", block: true, highlight: true, children: "Confirm" }),
    /* @__PURE__ */ (0, import_jsx_runtime37.jsx)(Button, { style: {
      marginTop: "24px"
    }, onClick: props.onClose, size: "lg", block: true, children: "Next Time" })
  ] }) });
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Components,
  Hooks,
  MatchProvider,
  useMatch
});
